src/Esfa.Vacancy.Register.Infrastructure/Caching/AzureRedisCacheService.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Threading.Tasks;
+using Esfa.Vacancy.Register.Domain.Interfaces;
+using Esfa.Vacancy.Register.Infrastructure.Settings;
+using Newtonsoft.Json;
+using SFA.DAS.NLog.Logger;
+using StackExchange.Redis;
+
+namespace Esfa.Vacancy.Register.Infrastructure.Caching
+{
+
+    public class AzureRedisCacheService : ICacheService
+    {
+        private readonly Lazy<ConnectionMultiplexer> _lazyConnection;
+        private readonly ILog _logger;
+
+        public AzureRedisCacheService(IProvideSettings settings, ILog logger)
+        {
+            _logger = logger;
+            var cacheConnection = settings.GetSetting(ApplicationSettingKeyConstants.CacheConnection);
+            _lazyConnection = new Lazy<ConnectionMultiplexer>(() => ConnectionMultiplexer.Connect(cacheConnection));
+        }
+
+        private ConnectionMultiplexer Connection => _lazyConnection.Value;
+
+        public async Task<T> CacheAsideAsync<T>(string key, Func<Task<T>> actionAsync, TimeSpan timeSpan)
+        {
+            var cache = Connection.GetDatabase();
+
+            var cachedValue = await cache.StringGetAsync(key);
```

Comment:
I haven;t really looked at this, but what happens if I want to get an int from the Redis cache?